ExtractUserDataActivity/handler.ts

```diff
@@ -371,107 +371,108 @@ export const queryAllUserData = (
   profileModel: ProfileModel,
   messageContentBlobService: BlobService,
   fiscalCode: FiscalCode
-): TaskEither<
+): TE.TaskEither<
   ActivityResultUserNotFound | ActivityResultQueryFailure,
   AllUserData
   // eslint-disable-next-line max-params
 > =>
-  // step 0: look for the profile
-  getProfile(profileModel, fiscalCode)
+  pipe(
+    // step 0: look for the profile
+    getProfile(profileModel, fiscalCode),
     // step 1: get messages, which can be queried by only knowing the fiscal code
-    .chain(profile =>
-      sequenceS(taskEither)({
+    TE.chain(profile =>
+      sequenceS(TE.ApplicativePar)({
         // queries all messages for the user
-        messages: messageModel
-          .findMessages(fiscalCode)
-          .chain(iterator =>
-            tryCatch(
+        messages: pipe(
+          messageModel.findMessages(fiscalCode),
+          TE.chain(iterator =>
+            TE.tryCatch(
               () => asyncIteratorToArray(iterator),
               toCosmosErrorResponse
             )
-          )
-          .map(flatten)
-          .mapLeft(_ =>
+          ),
+          TE.map(flatten),
+          TE.mapLeft(_ =>
             ActivityResultQueryFailure.encode({
               kind: "QUERY_FAILURE",
               query: "findMessages",
               reason: `${_.kind}, ${getMessageFromCosmosErrors(_)}`
             })
-          )
-          .foldTaskEither(
-            _ => fromLeft(_),
-            results =>
-              results.some(isLeft)
-                ? fromLeft(
-                    ActivityResultQueryFailure.encode({
-                      kind: "QUERY_FAILURE",
-                      query: "findMessages",
-                      reason: "Some messages cannot be decoded"
-                    })
-                  )
-                : fromEither(right(rights(results)))
           ),
-        profile: taskEither.of(profile)
+          TE.chainW(results =>
+            results.some(E.isLeft)
+              ? TE.left(
+                  ActivityResultQueryFailure.encode({
+                    kind: "QUERY_FAILURE",
+                    query: "findMessages",
+                    reason: "Some messages cannot be decoded"
+                  })
+                )
+              : TE.of(rights(results))
```

Comment:
maybe we can use `TE.fromPredicate`

Reply:
I struggled a bit with `fromPredicate`, then decided that a ternary would make the job. Also, `rights` creates an array of `Right`s instead of `Either`s